client/src/components/SharedGoal.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect } from "react";
+import styled from "styled-components/macro";
+import Container from "react-bootstrap/Container";
+import Profilephoto from "../assets/ProfilePhoto.png";
+import { FaUsers, FaCalendarAlt } from "react-icons/fa";
+import axios from "axios";
+import { useHistory } from "react-router-dom";
+
+const SharedGoal = ({ goal }) => {
+  const History = useHistory();
+  const joinGoal = (e) => {
+    e.preventDefault();
+    axios
+      .put("/goal/join", { goal_id: goal._id }, { withCredentials: true })
+      .then((res) => {
+        History.push("/goals");
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  };
+
+  return (
+    <Wrapper>
+      <Container>
+        <FlexDiv>
+          <Title>{goal ? goal.title : ""}</Title>
+          <FlexDiv>
+            <Profile src={Profilephoto} />
+            <UserName>Jane Doe</UserName>
+          </FlexDiv>
+        </FlexDiv>
+
+        <SubtitleDiv>
+          <Subtitle>{goal ? goal.category : ""}</Subtitle>
+          <Subtitle>{goal ? goal.period : ""}</Subtitle>
+        </SubtitleDiv>
+
+        <FlexDiv>
+          <SubtitleDiv>
+            <ItemIcon>
+              <FaUsers />
+            </ItemIcon>
+            <Stat>1223</Stat>
+            <ItemIcon>
+              <FaCalendarAlt />
+            </ItemIcon>
+            <Stat>{goal ? goal.timespan : ""} days left</Stat>
+          </SubtitleDiv>
+          <JoinButton onClick={joinGoal}>Join</JoinButton>
```

Comment:
Would be great if you can make it `joined` and non-clickable if user onGoingGoal already have this. Ignore if cannot handle :(